### DISCUSSION TITLE
[turbo.json/ui] add option for task to override global ui setting

### DISCUSSION THREAD

**Discussion by DerTimonius** [vercel/turborepo#9091] [Ideas]
### Goals

1. define UI mode more granularly with the option to override it in a task
2. 

### Background

I really love the TUI, but for some tasks it might make sense to use `stream` as a UI setting instead of the TUI.

An example would be unit tests in a single package with some interactivity (think Vitest) where in the TUI you would have to hit Enter and then interact with the output.

Adding the option to override the UI mode in specific tasks would allow for more granular use of the individual UI modes based on their use case.

It's not the end of the world if this wouldn't be implemented, but I think it would be nice to have.

### Proposal

Allow for a `turbo.json` like the following: 

```diff
{
  "$schema": "https://turborepo.org/schema.json",
  "ui": "tui",
  "tasks": {
    "start": {
      "cache": false
    },
    "test.unit": {
+	  "ui": "stream",
      "cache": true
    }
  }
}

```

**Comment by anthonyshew** [comment]
What behavior would you expect to see when running `turbo run start test:unit`?

**Reply by DerTimonius** [reply]
In this case, I would expect turbo to error due to the mismatch of UI modes. The error message would then also state why it did error.

Alternatively, the global UI configuration should be used (although this doesn't really make sense to me?).

What do you think of this?

**Reply by anthonyshew** [reply]
That sounds like a fairly unpleasant experience. If I run valid tasks, I would want them to get their work done.

Reading your proposal again, I just realized this line:
>  unit tests in a single package 

Does https://github.com/vercel/turborepo/discussions/8351 seem like it would suit your needs? The proposal there seems much more workable to me.

**Reply by DerTimonius** [reply]
That sounds like a different approach with pretty much the same result 🙂

**Reply by anthonyshew** [reply]
Would you say we could close this discussion in favor of that one, then?
